posthog/views.py

```diff
@@ -80,11 +80,39 @@ def stats(request):
 
 
 def robots_txt(request):
-    ROBOTS_TXT_CONTENT = (
-        "User-agent: *\nDisallow: /shared_dashboard/\nDisallow: /shared/"
-        if is_cloud()
-        else "User-agent: *\nDisallow: /"
-    )
+    # Block all on self-hosted instances
+    if not is_cloud():
+        return HttpResponse("User-agent: *\nDisallow: /", content_type="text/plain")
+
+    ROBOTS_TXT_CONTENT = """User-agent: *
+
+# Block shared paths
+Disallow: /shared_dashboard/
+Disallow: /shared/
+
+# Block URLs with sensitive query parameters
```

Comment:
TBH, any reason not to completely ignore paths with any query parameter?

Reply:
I don't see them, but we're definitely kinda blind. Is there any URL with a path parameter in archive.org that we would wanna keep?

Reply:
This isn't just for archive.org it's for all scrapers so if we have a dynamic landing page like /partner?p=name then this would not get scraped and those pages can be important. I'd rather not making sweeping changes that could cause issues in areas I'm not aware of.